src/core/thread/network_data_leader.hpp

```diff
@@ -245,6 +245,30 @@ class LeaderBase : public NetworkData
      */
     otError SetCommissioningData(const uint8_t *aValue, uint8_t aValueLength);
 
+    /**
+     * This method checks if the steering data allows a Joiner.
+     *
+     * @param[in]  aEui64             A reference to the Joiner's IEEE EUI-64.
+     *
+     * @retval OT_ERROR_NONE          The joiner is in the bloom filter.
+     * @retval OT_ERROR_INVALID_STATE No steering data present.
+     * @retval OT_ERROR_NOT_FOUND     The joiner is not in the bloom filter.
+     *
+     */
+    otError SteeringDataCheckJoiner(const Mac::ExtAddress &aEui64) const;
+
+    /**
+     * This method checks if the steering data allows a Joiner with a given discerner value.
+     *
+     * @param[in]  aDiscerner         A reference to the Joiner Discerner.
+     *
+     * @retval OT_ERROR_NONE          The joiner is in the bloom filter.
+     * @retval OT_ERROR_INVALID_STATE No steering data present.
+     * @retval OT_ERROR_NOT_FOUND     The joiner is not in the bloom filter.
+     *
+     */
+    otError SteeringDataCheckJoinerWithDiscerner(const MeshCoP::JoinerDiscerner &aDiscerner) const;
```

Comment:
We can consider using the same method name `SteeringDataCheckJoiner()` (overload the method name which is then determined based on it the param type being `JoinerDiscerner` or `Mac::ExtAddress`). We do similar with `Commissioner::AddJoiner()`.